art/attacks/poisoning/bullseye_polytope_attack.py

```diff
@@ -0,0 +1,422 @@
+from torch.optim.optimizer import Optimizer
+from tqdm.auto import trange
+
+from art.attacks.attack import PoisoningAttackWhiteBox
+from art.estimators import BaseEstimator, NeuralNetworkMixin
+from art.estimators.classification.classifier import ClassifierMixin
+from art.estimators.classification.pytorch import PyTorchClassifier
+
+if TYPE_CHECKING:
+    from art.utils import CLASSIFIER_NEURALNETWORK_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+class BullseyePolytopeAttack(PoisoningAttackWhiteBox):
+    """
+    Implementation of Bullseye Polytope Attack by Aghakhani, et. al. 2020.
+    "Bullseye Polytope: A Scalable Clean-Label Poisoning Attack with Improved Transferability"
+
+
+    This implementation is based on UCSB's original code here: https://github.com/ucsb-seclab/BullseyePoison
+
+    | Paper link: https://arxiv.org/abs/2005.00191
+    """
+
+    attack_params = PoisoningAttackWhiteBox.attack_params + [
+        "target",
+        "feature_layer",
+        "opt"
+        "max_iter",
+        "learning_rate",
+        "momentum",
+        "decay_iter",
+        "decay_coeff",
+        "epsilon",
+        "norm",
+        "dropout",
+        "endtoend",
+        "verbose",
+    ]
+
+    _estimator_requirements = (BaseEstimator, NeuralNetworkMixin, ClassifierMixin, PyTorchClassifier)
+
+    def __init__(
+            self,
+            classifier: Union["CLASSIFIER_NEURALNETWORK_TYPE", List["CLASSIFIER_NEURALNETWORK_TYPE"]],
+            target: np.ndarray,
+            feature_layer: Union[Union[str, int], List[Union[str, int]]],
+            opt: str = 'adam',
+            max_iter: int = 4000,
+            learning_rate: float = 4e-2,
+            momentum: float = 0.9,
+            decay_iter: Union[int, List[int]] = 10000,
+            decay_coeff: float = 0.5,
+            epsilon: float = 0.1,
+            dropout: int = 0.3,
+            net_repeat: int = 1,
+            endtoend: bool = True,
+            verbose: bool = True,
+    ):
+        """
+        Initialize an Feature Collision Clean-Label poisoning attack
+
+        :param classifier: The proxy classifiers used for the attack. Can be a single classifier or list of classifiers
+                           with varying architectures.
+        :param target: The target input(s) of shape (N, W, H, C) to misclassify at test time. Multiple targets will be
+                       averaged.
+        :param feature_layer: The name(s) of the feature representation layer(s).
+        :param opt: The optimizer to use for the attack. Can be 'adam', 'sgd', or 'signedadam'
+        :param max_iter: The maximum number of iterations for the attack.
+        :param learning_rate: The learning rate of clean-label attack optimization.
+        :param momentum: The momentum of clean-label attack optimization.
+        :param decay_iter: Which iterations to decay the learning rate.
+                           Can be a integer (every N interations) or list of integers [0, 500, 1500]
```

Comment:
```suggestion
                           Can be a integer (every N iterations) or list of integers [0, 500, 1500]
```

Reply:
Thanks for catching this. Corrected it now.